controllers/operator_controller.go

```diff
@@ -38,13 +40,15 @@ type OperatorReconciler struct {
 	client.Client
 	Scheme *runtime.Scheme
 
-	resolver *resolution.OperatorResolver
+	Resolver *resolution.OperatorResolver
```

Comment:
Does this need to be exported? Could we just create a GetEntity(id DeppyID) function for the OperatorReconciler type?

Reply:
Not sure if this is answering your second question, but for the first one it was done this way to allow the resolver to be set up separately from the controller in `main()`, which you can see in the prototype [here](https://github.com/varshaprasad96/operator-controller/blob/controller_prototype/main.go#L150). This seems to make it easier to set up the catalog source cache.

Reply:
That could be done with a `NewOperatorReconciler(c client, s scheme, r resolver) &OperatorReconciler` function, right? Point being that once the cache is set we shouldn't allow it to be manipulated directly.

Reply:
Well, avoid allowing people to manipulate the resolver that is.

Reply:
Yeah I see your point there, I'll update :+1:

Reply:
Updated